internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -767,6 +808,41 @@ func resourceCosmosDbAccountUpdate(d *pluginsdk.ResourceData, meta interface{})
 		return fmt.Errorf("making Read request on %s: %s", id, err)
 	}
 
+	oldCapabilities := make([]documentdb.Capability, 0)
+	oldCapabilitiesMap := make(map[string]interface{})
+	for _, v := range *resp.Capabilities {
+		c := documentdb.Capability{
+			Name: v.Name,
+		}
+		oldCapabilities = append(oldCapabilities, c)
+		oldCapabilitiesMap[strings.ToLower(*c.Name)] = *c.Name
+	}
+
+	if d.HasChange("capabilities") {
+
+		newCapabilities, newCapabilitiesMap := expandAzureRmCosmosDBAccountCapabilities(d)
+
+		err := checkCapabilitiesCanBeUpdated(kind, oldCapabilitiesMap, newCapabilitiesMap)
+		if err != nil {
+			return fmt.Errorf("updating CosmosDB Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		}
+
+		updateParameters := documentdb.DatabaseAccountUpdateParameters{
+			DatabaseAccountUpdateProperties: &documentdb.DatabaseAccountUpdateProperties{
+				Capabilities: newCapabilities,
+			},
+		}
+
+		future, err := client.Update(ctx, id.ResourceGroup, id.Name, updateParameters)
+		if err != nil {
+			return fmt.Errorf("updating CosmosDB Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		}
+
+		if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("waiting for the CosmosDB Account %q (Resource Group %q) to finish updating: %+v", id.Name, id.ResourceGroup, err)
+		}
```

Comment:
(as below) once this has been updated we should re-retrieve `read` so that it's up to date:

```suggestion
		future, err := client.Update(ctx, id.ResourceGroup, id.Name, updateParameters)
		if err != nil {
			return fmt.Errorf("updating CosmosDB Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
		}

		if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
			return fmt.Errorf("waiting for the CosmosDB Account %q (Resource Group %q) to finish updating: %+v", id.Name, id.ResourceGroup, err)
		}
		
		read, err := client.Get(ctx, id.ResourceGroup, id.Name)
		if err != nil {
			return fmt.Errorf("re-retrieving %s after update: %+v", *id, err)
		}
```

Reply:
Fixed.